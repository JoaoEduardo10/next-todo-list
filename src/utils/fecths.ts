import { json } from 'stream/consumers';

export type TDataProps = {
  email: string;
  password: string;
};
interface IUser {
  name: string;
  email: string;
  password: string;
}
const urlApi = process.env.NEXT_PUBLIC_URL_API as string;

const loginUser = async (data: TDataProps) => {
  const response = await fetch(`${urlApi}/login`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  const token = await response.json();

  return token;
};

const createUser = async (user: IUser) => {
  const { email, name, password } = user;

  const response = await fetch(`${urlApi}/users`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify({ email, password, name }),
  });

  return {
    ok: response.ok,
  };
};

export { loginUser, createUser };
