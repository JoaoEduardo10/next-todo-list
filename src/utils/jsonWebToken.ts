import jwt from 'jsonwebtoken';

export interface Ijwt {
  email: string;
  name: string;
  id: string | number;
}

export interface IjwtComplete {
  iat: number;
  exp: number;
}

export const compareJwt = (token: string): (Ijwt & IjwtComplete) | null => {
  try {
    const jwtHash = process.env.NEXT_PUBLIC_HASH_JWT as string;

    const varifyToken = jwt.verify(token, jwtHash);

    return varifyToken as Ijwt & IjwtComplete;
  } catch (error: any) {
    console.log(error.message);
    return null;
  }
};
