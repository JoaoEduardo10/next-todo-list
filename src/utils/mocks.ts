import configureStore from 'redux-mock-store';
import { TBoard } from '../types';
import { mockTask } from '../components/task/mock';

const mockStore = configureStore([]);

const mockBoards: TBoard[] = [
  { id: 'board1', boardName: 'Board 1', taskConnect: '123', userId: '2' },
  { id: 'board2', boardName: 'Board 2', taskConnect: '1234', userId: '2' },
];

const store = mockStore({
  boards: {
    allBoards: [
      { id: 'board1', boardName: 'Board 1' },
      { id: 'board2', boardName: 'Board 2' },
    ],
    actualBoard: { id: 'board1', boardName: 'Board 1' },
    actualBoardWithTasks: {
      boardName: 'Board 1',
      id: '1234',
      taskConnect: '1234',
      tasks: mockTask,
      userId: '1234',
    },
  },
});

export { store, mockBoards };
