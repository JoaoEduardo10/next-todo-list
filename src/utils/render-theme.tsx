import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { store as storeActual } from '../app/store';
import { Provider } from 'react-redux';

export const renderTheme = (
  children: React.ReactNode,
  store?: any,
): RenderResult => {
  return render(
    <Provider store={store ?? storeActual}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Provider>,
  );
};
