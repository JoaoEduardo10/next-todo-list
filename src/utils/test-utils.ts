import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '../../src/app/store';

export const createTestStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    devTools: false,
  });

  return store;
};
