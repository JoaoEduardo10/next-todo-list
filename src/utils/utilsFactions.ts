import { Tasks } from '../components/task/styles';
import { TTasks } from '../types';

const returnedSubTasksConcluded = (task: TTasks) => {
  const allSubTaskConcluded = task.subTasks.filter((total) => {
    return total.concluded == true;
  });

  return allSubTaskConcluded.length;
};

const returnTasksByStatus = (tasks: TTasks[]) => {
  const tasksConcluidedFilter = tasks.filter((task) => {
    return task.status == 'concluded';
  });

  const tasksPendingFilter = tasks.filter((task) => {
    return task.status == 'pending';
  });

  const tasksProgressFilter = tasks.filter((task) => {
    return task.status == 'progress';
  });

  return { tasksConcluidedFilter, tasksPendingFilter, tasksProgressFilter };
};

export { returnedSubTasksConcluded, returnTasksByStatus };
