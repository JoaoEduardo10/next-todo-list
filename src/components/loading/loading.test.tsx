import { Loading } from '.';
import { renderTheme } from '../../utils/render-theme';

describe('<Loading />', () => {
  it('should to match snapchot', () => {
    const { container } = renderTheme(<Loading />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
