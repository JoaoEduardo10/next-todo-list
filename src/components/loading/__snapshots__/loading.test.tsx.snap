// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should to match snapchot 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 8rem;
  height: 8rem;
  border: 0.8rem solid #827FD3;
  border-top: 0.8rem solid transparent;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation: loading 1s ease-in-out infinite;
  animation: loading 1s ease-in-out infinite;
}

.c2 {
  width: 4rem;
  height: 4rem;
  border: 0.6rem solid #1AEB47;
  border-bottom: 0.6rem solid transparent;
  border-radius: 50%;
  -webkit-animation: loading 1.5s ease-in infinite;
  animation: loading 1.5s ease-in infinite;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
</div>
`;
